imports/plugins/core/orders/client/components/invoiceActions.js

```diff
@@ -42,46 +56,161 @@ class InvoiceActions extends Component {
       </div>
     );
   }
+
   renderRefundForm() {
     const { adjustedTotal } = this.props;
 
     return (
-      <div className="invoice-summary">
-        <span  className="invoice-label refund">
-          <strong><Translation defaultValue="Refund" i18nKey="admin.invoice.refund"/></strong>
-        </span>
+      <div>
+        {this.props.hasRefundingEnabled &&
+          <div className="flex refund-container">
+            <div className="form-group order-summary-form-group">
+              <div className="invoice-details">
+                <NumericInput
+                  numericType="currency"
+                  value={this.state.value}
+                  maxValue={adjustedTotal}
+                  format={this.props.currency}
+                  classNames={{
+                    input: {
+                      amount: true
+                    }
+                  }}
+                  onChange={(event, data)=>{
+                    this.setState({
+                      value: data.numberValue
+                    });
+                  }}
+                />
+              </div>
 
-        <div className="invoice-details">
-          <span className="refundAmount">
-            <NumericInput
-              numericType="currency"
-              value="0"
-              maxValue={adjustedTotal}
-              classNames="amount"
-            />
-          </span>
-          <span className="refundSubmit">
-            <IconButton
-              bezelStyle={"flat"}
-              icon="fa fa-arrow-circle-o-right"
-              onClick={this.props.handleActionViewBack}
-            />
-            <span id="btn-refund-payment">-></span>
-          </span>
-        </div>
+            </div>
+
+            <Button
+              className="flex-item-fill refund-button"
+              type="button"
+              status="primary"
+              bezelStyle="solid"
+              disabled={this.props.isRefunding || this.state.value === 0}
+              onClick={(event) => {
+                this.props.handleRefund(event, this.state.value);
+                this.setState({
```

Comment:
Yup, coz the value on the refund input box is being set by the onChange function on this component instead of having the value change propagated from the parent component I have the value set in the same component. 

In short yes there is a reason. 
 --> Easier
 --> Have a central point where changes occur